scene/gui/graph_edit.cpp

```diff
@@ -1823,7 +1823,20 @@ HashMap<int, Vector<StringName>> GraphEdit::_layering(const HashSet<StringName>
 		}
 		if (!selected) {
 			current_layer++;
+			uint32_t previous_size_z = z.size();
 			_set_operations(GraphEdit::UNION, z, u);
+			if (z.size() == previous_size_z) {
+				WARN_PRINT("Graph contains cycle(s) so arranged one may be faulty.");
```

Comment:
We can change the warning to: `"Graph contains cycle(s). The cycle(s) will not be rearranged accurately."`